pkg/loki/modules.go

```diff
@@ -836,29 +837,21 @@ func (t *Loki) initIndexGatewayRing() (_ services.Service, err error) {
 	t.Cfg.StorageConfig.BoltDBShipperConfig.Mode = shipper.ModeReadOnly
 	t.Cfg.IndexGateway.Ring.KVStore.MemberlistKV = t.MemberlistKV.GetMemberlistKV
 	t.Cfg.IndexGateway.Ring.ListenPort = t.Cfg.Server.GRPCListenPort
-	ringCfg := t.Cfg.IndexGateway.Ring.ToRingConfig(t.Cfg.IndexGateway.Ring.ReplicationFactor)
-	reg := prometheus.WrapRegistererWithPrefix("loki_", prometheus.DefaultRegisterer)
 
-	logger := util_log.Logger
-	ringStore, err := kv.NewClient(
-		ringCfg.KVStore,
-		ring.GetCodec(),
-		kv.RegistererWithKVName(prometheus.WrapRegistererWithPrefix("loki_", reg), "index-gateway"),
-		logger,
-	)
-	if err != nil {
-		return nil, gerrors.Wrap(err, "kv new client")
+	managerMode := indexgateway.ClientMode
+	if t.isModuleActive(IndexGateway) && !t.isModuleActive(Ingester) {
```

Comment:
Why do we have to check for `!t.isModuleActive(Ingester)`? I think we can just do `if t.isModuleEnabled(IndexGateway)`?

Reply:
Oops, I mistakenly thought that the `isModuleEnabled` behavior is actually the one implemented by `isModuleActive`. Fixed!